### PR TITLE
fix(components): [notification] fix props.zIndex lose efficacy bug

### PR DESCRIPTION
**Pull request by Gnalvin**
option zIndex for notification can't effect

### CODE REVIEW

File: packages/components/notification/src/notification.vue [hunk 1]
```diff
@@ -84,7 +84,7 @@ const verticalProperty = computed(() =>
 const positionStyle = computed<CSSProperties>(() => {
   return {
     [verticalProperty.value]: `${props.offset}px`,
-    zIndex: currentZIndex.value,
+   zIndex: props.zIndex ? props.zIndex : currentZIndex.value,
```
Comment (jtejederabit) [inline packages/components/notification/src/notification.vue:87]: ```zIndex: props.zIndex ?? currentZIndex.value```

**Review by jtejederabit — COMMENTED** [review]
👍

File: packages/components/notification/src/notification.vue [hunk 2]
```diff
@@ -84,7 +84,7 @@ const verticalProperty = computed(() =>
 const positionStyle = computed<CSSProperties>(() => {
   return {
     [verticalProperty.value]: `${props.offset}px`,
-    zIndex: currentZIndex.value,
+    zIndex: props.zIndex ?? currentZIndex.value,
```
Comment (tolking) [inline packages/components/notification/src/notification.vue:87]: The default value of `zIndex` attribute  is `0`, so the `props.zIndex ?? currentZIndex.value` is always is `0`.

The default value of `zIndex` attribute is unreasonable, maybe we should remove it.

File: packages/components/notification/src/notification.ts [hunk 3]
```diff
@@ -107,8 +107,8 @@ export const notificationProps = buildProps({
    * @description initial zIndex
    */
   zIndex: {
-    type: Number,
-    default: 0,
+    type: definePropType<number | null>([Number, null]),
+    default: null,
   },
```
Comment (tolking) [inline packages/components/notification/src/notification.ts:112]:
```
zIndex: Number,
```

**Review by tolking — APPROVED** [review]

**Review by btea — APPROVED** [review]

**Pull request merged**